pandas/core/arrays/datetimelike.py

```diff
@@ -681,12 +681,13 @@ def value_counts(self, dropna=False):
 
         cls = type(self)
 
-        result = value_counts(values, sort=False, dropna=dropna)
+        result = value_counts(values.ravel("K"), sort=False, dropna=dropna)
```

Comment:
What's the reason for the ravel here? (value_counts is always done on a 1D array AFAIK)

Reply:
I agree this one cannot be ravel_compat (instead assert values.ndim == 1 here)

Reply:
updated+green